### PR TITLE
Adding Foundry support

### LINKED ISSUE #1767: Add Foundry support
We want Foundry to work along with Hardhat to be able to run unit tests with fuzzing.

Some useful links:
- https://book.getfoundry.sh/config/hardhat
- https://github.com/0xsequence/wallet-contracts/tree/v2
- https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3676

### PR DESCRIPTION
**Pull request by jubeira**
Closes #1767.

Adding support just for `solidity-utils` for now.

CI pending; we can either add Foundry as a separate task, or incorporate Foundry tests to the existing package tests.

### CODE REVIEW

File: pkg/solidity-utils/contracts/test/SimpleMock.sol [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+contract SimpleMock {
+    function pipe(uint256 a) public pure returns (uint256) {
+        return a;
```
Comment (nventuro) [inline pkg/solidity-utils/contracts/test/SimpleMock.sol:19]: We can delete this

File: pkg/solidity-utils/test/foundry/SimpleMock.t.sol [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import "forge-std/Test.sol";
+
+import "../../contracts/test/SimpleMock.sol";
+
+contract SimpleMockTest is Test {
+    SimpleMock simpleMock;
+
+    function setUp() public {
+        simpleMock = new SimpleMock();
+    }
```
Comment (nventuro) [inline pkg/solidity-utils/test/foundry/SimpleMock.t.sol:13]: Also this

File: pvt/common/skipFoundryTests.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { subtask } from 'hardhat/config';
+import { TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS } from 'hardhat/builtin-tasks/task-names';
+
+subtask(TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS).setAction(async (_, __, runSuper) =>
```
Comment (nventuro) [inline pvt/common/skipFoundryTests.ts:4]: You'll need to import this file in the base hardhat config

**Comment by nventuro** [comment]
>CI pending; we can either add Foundry as a separate task, or incorporate Foundry tests to the existing package tests.

Make it be a new task

File: pvt/common/skipFoundryTests.ts (hunk 3 above)
Comment (jubeira) [reply pvt/common/skipFoundryTests.ts:4]: Done.

Just as a sidenote, yarn / hardhat doesn't pick up these files either way if we put them inside `test/foundry`. Anyway I think it's good to have this filter.

**Review by nventuro — APPROVED** [review]
The foundry has been lit

**Comment by TomAFrench** [comment]
:its_happening:

**Pull request merged**